RecoBTag/PerformanceDB/test/BTagCalibrationStandalone.cc

```diff
@@ -122,18 +205,22 @@ BTagEntry::BTagEntry(const TH1* hist, BTagEntry::Parameters p):
     params.ptMax = axis->GetBinUpEdge(nbins);
   }
 
-  std::stringstream buff;
-  buff << "x<" << axis->GetBinLowEdge(1) << " ? 0. : ";  // default value
-  for (int i=1; i<nbins+1; ++i) {
-    char tmp_buff[100];
-    sprintf(tmp_buff,
-            "x<%g ? %g : ",  // %g is the smaller one of %e or %f
-            axis->GetBinUpEdge(i),
-            hist->GetBinContent(i));
-    buff << tmp_buff;
+  // balanced full binary tree height = ceil(log(2*n_leaves)/log(2))
+  // breakes even around 10, but lower values are more propable in pt-spectrum
+  if (nbins < 15) {
+    formula = th1ToFormulaLin(hist);
+  } else {
+    formula = th1ToFormulaBinTree(hist);
+  }
+
+  // compile formula to check validity
+  TF1 f1("", formula.c_str());
+  if (f1.IsZombie()) {
+std::cerr << "ERROR in BTagCalibration: "
+          << "Invalid histogram; formula does not compile (>150 bins?): "
+          << hist->GetName();
+throw std::exception();
```

Comment:
This is because the standalone code is auto-generated from the CMSSW code. In cmssw we use the cms::Exception, so it needs to be replaced for code that should be able to live outside of cmssw. The script is generate the standalone version is RecoBTag/PerformanceDB/python/generateStandaloneCode.py

The standalone version is meant for user who run their analyses without CMSSW. If not distributed within CMSSW, we would need to distribute the code over twiki, which is a bit cumbersome and easy to forget. Also the versions would be easily confused..